File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int currxor = 0;
        vector<int> ans;
        for(int & i:arr){
            currxor = currxor^i;
            i = currxor;
        }
        for(int i = 0; i < queries.size(); i++){
            int start = queries[i][0];
            int end = queries[i][1];
            if (start == 0) ans.push_back(arr[end]);
            else ans.push_back(arr[end]^arr[start-1]);
        }
        return ans;
    }
};

File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int currXor = 0;
        //prefix sum
        for(int i = 0; i < arr.length; i++){
            currXor ^= arr[i];
            arr[i] = currXor;
        }
        int[] ans = new int[queries.length];
        for(int i = 0; i < ans.length; i++){
            if(queries[i][0] == 0) ans[i] = arr[queries[i][1]];
            else ans[i] = arr[queries[i][0]-1]^arr[queries[i][1]];
        }
        return ans;
    }
}